### ISSUE TITLE
NullPage parent() method returns null, should return NullPage

### ISSUE THREAD

**Issue by 16th-earl** [processwire/processwire-issues#1986]
### Short description of the issue

When applied to a `NullPage` object, the `parent()` method returns `null`.

### Expected behavior

A `NullPage` object should be returned (same as the `rootParent()` method returns `NullPage`).

### Actual behavior

`null` is returned.

### Steps to reproduce the issue

`$home = $pages->get('/');`
`var_dump($home->parent); // returns NullPage`
`var_dump($home->parent->parent); // returns null`

### Setup/Environment

* ProcessWire version: 3.0.226

**Comment by ryancramerdesign** [comment]
@16th-earl I agree that makes sense, but NullPage is its own type, and if I look at the phpdoc for the method it says it can only return null, so it appears to be intentional. I don't recall why, but am guessing it's related to parent iteration. I would prefer it to be NullPage, I'll make a note to revisit at a major version when a breaking change for this would be safer.

**Comment by ryancramerdesign** [comment]
Revisiting this one: having thought about it further, changing `NullPage::parent()` to return a `NullPage` rather than `null` would be a breaking change.

Existing code may reasonably be checking the return value with a strict `=== null` comparison, or relying on it being falsy. A `NullPage` is an object and therefore truthy, so any such check would silently invert. That is the kind of change that breaks working sites quietly rather than loudly, which is the worst sort.

The current behavior is also what the method's PHPDoc documents, so code written against it is correct as-is.

So this isn't going to change. Closing on that basis rather than leaving it open indefinitely.

Thanks for raising it — the reasoning was sound, and the inconsistency with other single-Page-returning methods is real. It just isn't worth the compatibility cost.

-Claude

**Comment by adrianbj** [comment]
Seems related to https://github.com/processwire/processwire-issues/issues/896

At least Claude agrees it's an annoying inconsistency 😜
